### PR TITLE
Extract schema loading, linking, pruning into its own module.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: wire-schema/src/test/java/com/squareup/wire/schema/OptionsTest.java [hunk 1]
```diff
@@ -13,8 +13,9 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.squareup.wire;
+package com.squareup.wire.schema;
 
+import com.squareup.wire.schema.Options;
```
Comment (danrice-square) [inline wire-schema/src/test/java/com/squareup/wire/schema/OptionsTest.java]: Why an import from the same package?

File: wire-schema/src/test/java/com/squareup/wire/schema/TypeNameTest.java [hunk 2]
```diff
@@ -13,8 +13,9 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.squareup.wire;
+package com.squareup.wire.schema;
 
+import com.squareup.wire.schema.Type;
```
Comment (danrice-square) [inline wire-schema/src/test/java/com/squareup/wire/schema/TypeNameTest.java]: Why an import from the same package?

**Comment by danrice-square** [comment]
LGTM

File: wire-schema/src/test/java/com/squareup/wire/schema/OptionsTest.java (hunk 1 above)
Comment (swankjesse) [inline wire-schema/src/test/java/com/squareup/wire/schema/OptionsTest.java]: Caught in refactoring. Fixed.

File: wire-schema/src/test/java/com/squareup/wire/schema/TypeNameTest.java (hunk 2 above)
Comment (swankjesse) [inline wire-schema/src/test/java/com/squareup/wire/schema/TypeNameTest.java]: Caught in refactoring. Fixed.

**Pull request merged**